hazelcast/src/main/java/com/hazelcast/spi/impl/proxyservice/impl/ProxyRegistry.java

```diff
@@ -91,16 +90,8 @@ boolean contains(String name) {
      * @param result The ProxyInfo of all proxies in this registry.
      */
     public void getProxyInfos(Collection<ProxyInfo> result) {
-        for (Map.Entry<String, DistributedObjectFuture> entry : proxies.entrySet()) {
-            final DistributedObjectFuture future = entry.getValue();
-            if (!future.isSet()) {
```

Comment:
that's a good point. 

what's the current behaviour? when a future is already set (=the loading was completed) then it will send the proxy to the newly joined node and the loading will be triggered anyway? so the only difference is when then `future.set()` returns false (=it's still loading). Is this correct? 

I think all actions in the proxy initialization should be re-entrant.